### PR TITLE
[build-script] Add support for building, installing, and testing swift-driver

### PR DESCRIPTION
**Pull request by DougGregor**
Add build-script command-line options for building, installing, and
testing swift-driver:
* `--swift-driver` will build the Swift driver. If testing is enabled,
  it will be tested as well
* `--install-swift-driver` will install the `swift-driver` and
  `swift-help` executables in the toolchain.
* `--skip-test-swift-driver` will disable testing of the Swift driver
when other tests are being run.

The Swift driver depends on SwiftPM to build; it is recommended that
you use `--infer` to get the appropriate dependencies built.

Note that this option does not yet replace the existing Swift driver
executables (`swiftc`, `swift`) with the new driver, nor does it
install the SwiftDriver library for use elsewhere.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

File: utils/swift_build_support/swift_build_support/products/swiftdriver.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        return False
+
+    def should_build(self, host_target):
+        return self.args.build_swift_driver
+
+    @classmethod
+    def get_dependencies(cls):
+        return [cmark.CMark,
+                llvm.LLVM,
+                libcxx.LibCXX,
```
Comment (compnerd) [inline utils/swift_build_support/swift_build_support/products/swiftdriver.py:46]: Is libc++ a strict dependency?  What about non-Darwin targets?

Comment (DougGregor) [reply utils/swift_build_support/swift_build_support/products/swiftdriver.py:46]: I cargo-culted this from SwiftPM's dependencies. @gottesmm do you recall why we have the libc++ dependency here?

Comment (gottesmm) [reply utils/swift_build_support/swift_build_support/products/swiftdriver.py:46]: This is b/c I used the most conservative set of dependencies that build-script had. If you don't need it (or these other projects don't) then I would just remove it.

I think of it as a base part of the just built toolchain like clang/compiler-rt. So I imagine it doesn't hurt anything.

Comment (gottesmm) [reply utils/swift_build_support/swift_build_support/products/swiftdriver.py:46]: IOW, build-script had a linear ordering that never changed. I created the simplest correct dependency chain by (ignoring lldb) making all products depend on all other products before them.

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Comment by DougGregor** [comment]
@swift-ci please test Windows

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Pull request merged**